woocommerce/woocommerce
# 31451 - Filter Out Disabled Attributes in Filter Products Widget

***WIP: Opening PR for Discussion***

Addresses #31451.

I'd like to request feedback on this approach to solving this issue.

### Changes proposed in this Pull Request:

In cases where a variation attribute is being used, but is not actually "Enabled" on **any** product, the attribute still displays in the Layered Nav widget (also known as the "Filter Products by Attribute" widget).  

Note that using "hide_empty" in `get_terms()` is unable to detect Disabled attributes because the attribute is technically still assigned to the product.  

This PR addresses this by checking each term ("attribute"), fetching up to 250 posts with that term and checking to see if any of them actually have the variation using that attribute Enabled.  

As this method uses tax_queries, this code uses transients to store query results.  For testing, the transient length is set to just 10 seconds.  This would be adjusted after code review based on the reviewer's recommendations.


### How to test the changes in this Pull Request:

1. Create a variable product and at least one variation with a newly-created attribute.
2. Disable that variation by ensuring the "Enabled" checkbox is empty.
3. Add the "Filter Products by Attribute" widget to any widget area.
4. Check to see that the unused ("not enabled anywhere") attribute does not show up in the filters.

There are more detailed instructions on reproducing these conditions in the original Issue.

### Other notes:

- This bug will also require adjusting the main query on the attribute term Archive, as products with this attribute assigned to a variation will still  display, even if the variation itself is disabled.
- Unit tests have not been run on this code yet.

---



### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### FOR PR REVIEWER ONLY:
